src/test/regress/sql/node_conninfo_reload.sql

```diff
@@ -205,4 +205,30 @@ show citus.node_conninfo;
 -- Should work again
 ALTER TABLE test ADD COLUMN e INT;
 
+-- show that we allow providing "host" param via citus.node_conninfo
+ALTER SYSTEM SET citus.node_conninfo = 'sslmode=require host=nosuchhost';
```

Comment:
If this feature is to avoid dns resolution for the hostname `pg_dist_node.nodename` could be set to an ip addr.
If it is to allow `hostaddr ` and  `host` parameters together for avoiding the resolution while being able to use hostname,   hostaddr and hostname  should better be maintained in pg_dist_node. 

As` citus.node_conninfo` is common for all nodes, it looks off and limited to include target host related info here.

Reply:
yes, it's to avoid allowing dns resolution (and/or setting up DNS names for each host in the cluster). This already works currently when using IPs in the hostname. The only use of setting `host` is that you can then use `sslmode=verify-full` and it will validate that the hostname matches the certificate provided by the node you're connecting too.

I agree that it would be more flexible to make this a per-node setting, but that requires SQL changes. And we'd like to backport this change. And in many setups a different hostname for TLS validation is actually not needed. The reason for that is query-from-any node: With query-from-any-node all nodes usually have a certificate that is valid for the same "cluster hostname", either using a wildcard cert or a Subject Alternative Name (SAN). Because if you load balance across nodes you don't know which node you're connecting to, but you still want TLS validation to do it's job. So with this change you can use this same "cluster hostname" for TLS validation within the cluster. Obviously this means you don't validate that you're connecting to a particular node, just that you're connecting to one of the nodes in the cluster, but that should be fine from a security perspective (in most cases).

@onurctirtir probably good to add this explanation to the PR description/code comment.

Reply:
done, added to PR description